### PR TITLE
HECv1 protocol json format support for un-marshalling

### PR DESCRIPTION
**Pull request by jgheewala**

### CODE REVIEW

File: protocol/signalfx/format/signalfx_format_easyjson.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ var (
 	_ easyjson.Marshaler
 )
 
-func easyjson3b0ebf6aDecodeGithubComSignalfxGatewayProtocolSignalfxFormat(in *jlexer.Lexer, out *JSONEventV2) {
+func easyjson3b0ebf6aDecodeGithubComSignalfxIngestProtocolsProtocolSignalfxFormat(in *jlexer.Lexer, out *JSONLogHECV1List) {
```
Comment (sk-sfx) [inline protocol/signalfx/format/signalfx_format_easyjson.go:22]: Is this an auto-generated file?

File: protocol/signalfx/format/signalfx_format.go [hunk 2]
```diff
@@ -98,3 +98,19 @@ type InputSpan struct {
 // InputSpanList is an array of InputSpan pointers
 //easyjson:json
 type InputSpanList []*InputSpan
+
+// JSONLogHECV1 decodes enterprise HEC format, details: https://docs.splunk.com/Documentation/SplunkCloud/latest/Data/FormateventsforHTTPEventCollector#Event_metadata
+//easyjson:json
+type JSONLogHECV1 struct {
+	Time       *int64                 `json:"time,omitempty"`
```
Comment (sk-sfx) [inline protocol/signalfx/format/signalfx_format.go:105]: Do we need to make some fields not have omitempty?

Comment (jgheewala) [reply protocol/signalfx/format/signalfx_format.go:105]: based on Document 

> The keys in Event Metadata are all optional.

File: protocol/signalfx/format/signalfx_format_easyjson.go (hunk 1 above)
Comment (jgheewala) [reply protocol/signalfx/format/signalfx_format_easyjson.go:22]: this file is auto-generated

File: protocol/signalfx/format/signalfx_format.go [hunk 3]
```diff
@@ -98,3 +98,19 @@ type InputSpan struct {
 // InputSpanList is an array of InputSpan pointers
 //easyjson:json
 type InputSpanList []*InputSpan
+
+// JSONLogHECV1 decodes enterprise HEC format, details: https://docs.splunk.com/Documentation/SplunkCloud/latest/Data/FormateventsforHTTPEventCollector#Event_metadata
+//easyjson:json
+type JSONLogHECV1 struct {
```
Comment (sk-sfx) [inline protocol/signalfx/format/signalfx_format.go:104]: Can we name it HECLogV1 for consistency?

File: protocol/signalfx/format/signalfx_format.go (hunk 2 above)
Comment (tigrannajaryan) [inline protocol/signalfx/format/signalfx_format.go:105]: I think HEC allows fractional timestamps as strings. int64 will prevent that.

**Pull request merged**